src/core/ext/filters/client_channel/retry_filter.h

```diff
@@ -19,12 +19,98 @@
 
 #include <grpc/support/port_platform.h>
 
+#include <limits.h>
+#include <stddef.h>
+
+#include <new>
+
+#include "absl/types/optional.h"
+
+#include <grpc/event_engine/event_engine.h>
+#include <grpc/grpc.h>
+#include <grpc/support/log.h>
+
+#include "src/core/ext/filters/client_channel/client_channel.h"
+#include "src/core/ext/filters/client_channel/retry_service_config.h"
+#include "src/core/ext/filters/client_channel/retry_throttle.h"
+#include "src/core/lib/channel/channel_args.h"
 #include "src/core/lib/channel/channel_fwd.h"
 #include "src/core/lib/channel/channel_stack.h"
+#include "src/core/lib/channel/context.h"
+#include "src/core/lib/debug/trace.h"
+#include "src/core/lib/gpr/useful.h"
+#include "src/core/lib/gprpp/ref_counted_ptr.h"
+#include "src/core/lib/iomgr/error.h"
+#include "src/core/lib/transport/transport.h"
+
+extern grpc_core::TraceFlag grpc_retry_trace;
 
 namespace grpc_core {
 
-extern const grpc_channel_filter kRetryFilterVtable;
+class RetryFilter {
+ public:
+  static const grpc_channel_filter kVtable;
+
+  static grpc_error_handle Init(grpc_channel_element* elem,
```

Comment:
I'd prefer not to expose anything outside of this class that we don't actually need to.  I think basically everything except `kVtable` can be private.

Note that the `RetryFilterLegacyCallData` class can still be nested inside of this class and be private, even if it's defined in a separate file.  That should eliminate the need for many of the new accessor methods here.

Reply:
I'm really not comfortable with exposing all of these internal implementation details; I think exposing them is more likely to cause problems than having them in another file.  I understand the desire to not split a class between two files, but we don't actually have any requirement to do that in the first place.  If you want to split it, I'm fine with that, as long as we keep the internal implementation details private.  If you're not comfortable with that, then I'd prefer to just keep everything in one file the way it is now.

Reply:
What potential change that gets past code review are you trying to prevent here?

Reply:
I think that if a class exposes a bunch of state, a reviewer can reasonably assume that it was intentional, and they may not realize that it's not a good idea to use that state externally.  I think it's a more obvious red flag if a PR is changing a class to expose state that was not previously exposed.

Reply:
So my thinking is maybe a comment `TODO(xxx): these methods public to support the transition from filter stack -> promises, cleanup after`, coupled with the large amount of work that's required to go dig the retry filter out of a channel stack to call it (that loop should always be a red flag) probably solves this?